pkg/ema.py

```diff
@@ -0,0 +1,113 @@
+import pymysql
+import time
+
+from .log import log
+
+# get a mysql db connection
+def get_conn():
+  return pymysql.connect('localhost', 'root', '', 'ema')
+
+# send request to ema and poll results
+def call_ema(id, suid):
+  err = None
+  empathid = None
+  time_received = 'NA'
+  response = -1.0
+
+  # the time must be between 8:00 am and 12:00 am
+  if datetime.now().hour < 8:
+    return
+
+  action = str(action)
+
+  # time sending the prequestion
+  start_time = time.time()
+  # date and time format of the time the prequestion is sent
+  time_sent = str(datetime.fromtimestamp(int(start_time)))
+
+  # items needed in url
+  empathid = '999|' + str(int(time.time()))
+
+  phone_url = 'http://191.168.0.106:2226'
+  server_url = 'http://191.168.0.107/ema/ema.php'
+  androidid = 'db7d3cdb88e1a62a'
+  alarm = 'true'
+
+  # sending action to phone
+  url_dict = {
+    'id': str(id),
+    'c':'startsurvey',
+    'suid': str(suid),
+    'server': server_url,
+    'androidid': androidid,
+    'empathid': empathid,
+    'alarm': alarm
+  }
+  q_dict_string = urllib.parse.quote(json.dumps(url_dict), safe=':={}/')  # encoding url quotes become %22
+  url = phone_url + '/?q=' + q_dict_string
+
+  try:
+    send = urllib.request.urlopen(url)
+  except http.client.BadStatusLine: # EMA return non-TCP response
+    pass
+
+  # connect to database for logging
+  try:
+    db = get_conn()
+    cursor = db.cursor()
+
+    insert_query = "INSERT INTO reward_data(empathid,TimeSent,RecommSent,TimeReceived,Response,Uploaded) \
```

Comment:
always create a logging record about what we have sent
and later update it with response in the polling

Reply:
got it!